src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java

```diff
@@ -0,0 +1,304 @@
+package au.org.emii.geoserver.wms;
+
+import au.com.bytecode.opencsv.CSVReader;
+import org.apache.commons.io.IOUtils;
+import org.geotools.util.logging.Logging;
+import org.joda.time.DateTime;
+import org.json.JSONObject;
+
+import java.io.*;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import java.net.URL;
+import java.util.*;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class Ncwms {
+    static Logger LOGGER = Logging.getLogger("au.org.emii.geoserver.extensions.filters");
+
+    private static String timeFieldName = "time";
+    private static String urlFieldName= "file_url";
+
+    public static Map<String, String> urlSubstitutions = new HashMap<String, String>();
+    private static String wfsServer = "http://localhost:8080/geoserver/ows";
```

Comment:
That's the default value for what we query, it is override-able using spring.

In the near future we will use Geoserver API directly, which returns a `FeatureCollection`. Then also no need for all the CSV stuff.